### PR TITLE
Allow paginated search using ES's search_after in SearchResult:fetchNext

### PR DESCRIPTION
**Pull request by samniisan**
## Introduces
- Way to use ES's search_after if conditions are met when calling SearchResult:fetchNext (better than using scrolls, bypasses the default search limit count)
- New unit tests to cover SearchResult functionalities

## Related issue
https://github.com/kuzzleio/kuzzle/issues/856

## Boyscoot
- Updates README.md (typos / various)
- PHPDoc fixes, invalid variable names, code cleanup

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,6 +34,30 @@ This SDK can be used in any project using composer:
 composer require kuzzleio/kuzzle-sdk
 ```
 
+## Basic usage
```
Comment (stafyniaksacha) [inline README.md:37]: :+1:

**Review by stafyniaksacha — APPROVED** [review]

**Comment by scottinet** [comment]
I've raised a question on the Javascript SDK implementation of this feature: https://github.com/kuzzleio/sdk-javascript/pull/233#pullrequestreview-44849333

It may be a good idea to take a look at it before merging this PR, to check if this is worth considering or not.

File: src/Util/SearchResult.php [hunk 2]
```diff
@@ -135,7 +135,21 @@ public function fetchNext()
...
+                return null;
+            }
+
+            $filters = $this->filters;
+            $filters['search_after'] = array_map(function ($sortRule) {
+                return end($this->documents)->getContent()[key($sortRule)];
+            }, $this->filters['sort']);
+
+            $searchResult = $this->collection->search($filters, $this->options);
+        } else if (array_key_exists('scrollId', $this->options)) {
```
Comment (benoitvidis) [inline src/Util/SearchResult.php]: if a scroll was explicitly requested, it should not be bypassed.

File: src/Util/SearchResult.php [hunk 3]
```diff
@@ -135,7 +135,21 @@ public function fetchNext()
     {
         $searchResult = null;
 
-        if (array_key_exists('scrollId', $this->options)) {
+        if (array_key_exists('size', $this->options)
+            && array_key_exists('sort', $this->filters)
+            && !array_key_exists('from', $this->options)) {
```
Comment (benoitvidis) [inline src/Util/SearchResult.php]: Why not use search after if `from` is set? Removing it from the search after request should still be consistent.

**Review by benoitvidis — CHANGES_REQUESTED** [review]

**Comment by samniisan** [comment]
PR updated, scroll is now prioritized as long as the user specifies a scroll param.

**Review by dbengsch — APPROVED** [review]

**Review by ballinette — APPROVED** [review]

**Pull request merged**